diff --git a/CMakeLists.txt b/CMakeLists.txt
index 433f22dc7..f235c888b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,5 +39,7 @@ az_vcpkg_integrate()
 
 # Project definition
 project(azure-sdk LANGUAGES CXX C)
-set(CMAKE_CXX_STANDARD 14)
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 set(CMAKE_CXX_STANDARD_REQUIRED True)
@@ -179,12 +179,12 @@ endif()
 # sub-projects
 add_subdirectory(sdk/core)
 add_subdirectory(sdk/appconfiguration)
-add_subdirectory(sdk/attestation)
+#add_subdirectory(sdk/attestation)
 # AMQP doesn't work for UWP yet, and eventhubs depends on AMQP, so we cannot include eventhubs on UWP.
 if (NOT DISABLE_AMQP)
     add_subdirectory(sdk/eventhubs)
 endif()
-add_subdirectory(sdk/identity)
+#add_subdirectory(sdk/identity)
 add_subdirectory(sdk/keyvault)
 add_subdirectory(sdk/storage)
 add_subdirectory(sdk/template)
diff --git a/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp b/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp
index 2ffd76cee..359412574 100644
--- a/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp
+++ b/sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp
@@ -91,6 +91,8 @@ namespace Azure { namespace Core { namespace Http {
      */
     Azure::Nullable<std::string> Proxy;
 
+    Azure::Nullable<std::string> NoProxy;
+
     /**
      * @brief Username to be used for proxy connections.
      *
diff --git a/sdk/core/azure-core/src/http/curl/curl.cpp b/sdk/core/azure-core/src/http/curl/curl.cpp
index b9180c119..33073c94d 100644
--- a/sdk/core/azure-core/src/http/curl/curl.cpp
+++ b/sdk/core/azure-core/src/http/curl/curl.cpp
@@ -764,7 +764,7 @@ inline std::string CurlSession::GetHTTPMessagePreBody(Azure::Core::Http::Request
   // If we're not using a proxy server, *or* the URL we're connecting uses HTTPS then
   // we want to send the relative URL (the URL without the host, scheme, port or authn).
   // if we ARE using a proxy server and the request is not encrypted, we want to send the full URL.
-  if (!m_httpProxy.HasValue() || request.GetUrl().GetScheme() == "https")
+  if (!m_connection->IsUsedProxy() || request.GetUrl().GetScheme() == "https")
   {
     url = "/" + request.GetUrl().GetRelativeUrl();
   }
@@ -2496,6 +2496,17 @@ CurlConnection::CurlConnection(
   /******************** Curl handle options apply to all connections created
    * The keepAlive option is managed by the session directly.
    */
+  if (options.NoProxy)
+  {
+    if (!SetLibcurlOption(m_handle, CURLOPT_NOPROXY, options.NoProxy->c_str(), &result))
+    {
+      throw Azure::Core::Http::TransportException(
+          _detail::DefaultFailedToGetNewConnectionTemplate + hostDisplayName
+          + ". Failed to set noproxy to:" + options.NoProxy.Value() + ". "
+          + std::string(curl_easy_strerror(result)));
+    }
+  }
+
   if (options.Proxy)
   {
     if (!SetLibcurlOption(m_handle, CURLOPT_PROXY, options.Proxy->c_str(), &result))
diff --git a/sdk/core/azure-core/src/http/curl/curl_connection_private.hpp b/sdk/core/azure-core/src/http/curl/curl_connection_private.hpp
index 60ccb558f..6a86cf8d0 100644
--- a/sdk/core/azure-core/src/http/curl/curl_connection_private.hpp
+++ b/sdk/core/azure-core/src/http/curl/curl_connection_private.hpp
@@ -131,6 +131,12 @@ namespace Azure { namespace Core {
        */
       virtual bool IsExpired() = 0;
 
+      /**
+       * @brief Checks whether this CURL connection is used proxy.
+       *
+       */
+      virtual bool IsUsedProxy() = 0;
+
       /**
        * @brief This function is used when working with streams to pull more data from the wire.
        * Function will try to keep pulling data from socket until the buffer is all written or until
@@ -236,6 +242,13 @@ namespace Azure { namespace Core {
         return connectionOnWaitingTimeMs.count() >= _detail::DefaultConnectionExpiredMilliseconds;
       }
 
+      bool IsUsedProxy() override
+      {
+        long used = 0;
+        CURLcode res = curl_easy_getinfo(m_handle.get(), CURLINFO_USED_PROXY, &used);
+        return (res == CURLE_OK) && used;
+      }
+
       /**
        * @brief This function is used when working with streams to pull more data from the wire.
        * Function will try to keep pulling data from socket until the buffer is all written or until
